### PR TITLE
Manifest rebase OIDC authservice

### LINKED ISSUE #21: Argoflow manifest - oidc-authservice
Compare our manifest and kubeflow's, with careful checking for any customization.
Update the argoflow-azure applications in distribution
Test
Duplicate the changes in the template.

### PR DESCRIPTION
**Pull request by Jose-Matsuda**
Would close #21 
I left a _lot_ of comments for your viewing pleasure. These can be cleaned up / deleted before merging.

It synchs up fine but does not progress into healthy because of `statefulset.yaml` but I believe its because we need to set up an `imagePullSecret`
I tried following [this guide](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-kubernetes) to try and create my own personal secret but it seems to fail for me. 

When everything is a-ok and after comments have been cleaned up after you have viewed them I will duplicate in the `template` folder.

### CODE REVIEW

File: distribution/oidc-authservice/statefulset.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      # this secret must exist in our deployment.
+      # would need to create our own secrets to put whenever we do something locally.
+      volumes:
+        - name: data
+          persistentVolumeClaim:
+              claimName: authservice-pvc #value comes from pvc.yaml in upstream, unchanged in 1.2 -> 1.3
+
+# 1.3 https://github.com/kubeflow/manifests/blob/v1.3-branch/common/oidc-authservice/base/statefulset.yaml
+# 1.2 https://github.com/kubeflow/manifests/blob/v1.2-branch/istio/oidc-authservice/base/statefulset.yaml
+# Is very similar to 1.2 with a few changes and the imagePullSecrets
```
Comment (wg102) [inline distribution/oidc-authservice/statefulset.yaml]: For reference?

File: kind/metallb/configmap.yaml [hunk 2]
```diff
@@ -9,4 +9,4 @@ data:
     - name: default
       protocol: layer2
       addresses:
-      - 172.20.255.200-172.20.255.250
```
Comment (wg102) [inline kind/metallb/configmap.yaml]: I would suggest avoiding changes in this file, since it depends on the developer's IP.

**Review by wg102 — CHANGES_REQUESTED** [review]
I've manage to get it up, and create a notebook. 
As you mentionned, there is a lot of comments. I would suggest keeping them somewhere, possibly the PR? Since they seem to explain a lot.

Finally, need all the changes in the other folder as well.

**Comment by Jose-Matsuda** [comment]
# File specific changes / comments:

These are comments that I had in the files about the changes required

---------------------------------------------------------------
`distribution/oidc-authservice/configmap.yaml` 
```
#Name in metadata must match whats in statefulset.yaml under envFrom/configMapRef/name
#Lines 7,12,14-19 are changed to make use of envFrom instead of having values in statefulset.yaml and have it in 1 spot.
#Values may differ a bit from 1.3 as I kept what we had for values in statefulset.yaml (userid_claim and oidc_scopes for example)
# are these case sensitive / need double quotes?
#changed 14 and 15 to have underscores instead of dashes, 
#Renames to have correct key value: 7 application_secret changed to client_secret, 11 oidc_redirect_uri to redirect_url
#Should override whatever is in the oidc-authservice-parameters configmap
# We don't have control over upstream params.env so I think a strategic merge is the play here
```
------------------------------------------------------------------
` distribution/oidc-authservice/envoy-filter.yaml`
There were indenting issues here that once fixed worked just fine. 
Additionally there was a sub-filter bit that I commented out as it wasn't necessary / not being used
```
# Notes 
# Taken from https://github.com/StatCan/aaw-kubeflow-manifest/blob/master/kustomize/oidc-authservice/envoy-filter.yaml
# 1.3 https://github.com/kubeflow/manifests/blob/v1.3-branch/common/oidc-authservice/base/envoy-filter.yaml
# 1.2 https://github.com/kubeflow/manifests/blob/v1.2-branch/istio/oidc-authservice/base/envoy-filter.yaml

# Looking at whats in 1.3 it seems fine, the variables in the server_uri come from the kustomization.yaml
# The one bit I'm unsure about is on line 42 about exact: kubeflow-userid but I think it is fine. 
## ^ if it's related to our change in configmap.yaml for the userid_claim then it might be funky
# I dont fully understand the config in 1.3 vs 1.2 so we just pray i guess 
# ### Getting an error of 
#error validating data: ValidationError(EnvoyFilter.spec.configPatches[0]): 
#unknown field "listener" in io.istio.networking.v1alpha3.EnvoyFilter.spec.configPatches
```
-------------------------------------------------------------------------
`distribution/oidc-authservice/kustomization.yaml`
```
# Notes
#### params.yaml
# Both 1.2 and 1.3 have this "params.yaml"
# The paths under 'varReference' point to paths depending on the 'kind' so for kind EnvoyFilter its in the envoy-filter.yaml
# In 1.3 the StatefulSet got removed (seems to be fine).
# No more spec/workloadLabels/istio in envoyfilter is ok (as long as this value is not used somewhere which it should not). 
# Is used in the kustomization.yaml for 'configurations'
#### kustomization.yaml
# theres a difference that 1.3 has a secretGenerator bit that uses secret_params.env which is whats in our configmap.yaml
# might need to look at that.
# From 'vars' there are a LOT of objref ConfigMap names in there. It's essentially whatever is in params.env
# Each of these have a fieldRef/fieldPath bit that I'm unsure about as well.
```
--------------------------------------------------------------------
` distribution/oidc-authservice/statefulset.yaml`
```
     imagePullSecrets: 
        - name: k8scc01covidacr-registry-connection
      # this secret must exist in our deployment.
      # would need to create our own secrets to put whenever we do something locally.
      volumes:
        - name: data
          persistentVolumeClaim:
              claimName: authservice-pvc #value comes from pvc.yaml in upstream, unchanged in 1.2 -> 1.3

# 1.3 https://github.com/kubeflow/manifests/blob/v1.3-branch/common/oidc-authservice/base/statefulset.yaml
# 1.2 https://github.com/kubeflow/manifests/blob/v1.2-branch/istio/oidc-authservice/base/statefulset.yaml
# Is very similar to 1.2 with a few changes and the imagePullSecrets
```

**Pull request merged**